src/utils/spp.spec.ts

```diff
@@ -0,0 +1,59 @@
+
+import assert from 'assert';
+import sinon from 'sinon';
+import { spp } from './spp.js';
+import { sinonUtil } from './sinonUtil.js';
+import request from '../request.js';
+
+describe('utils/spp', () => {
+  const siteUrl = 'https://contoso.sharepoint.com';
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+  });
+
+  it('calls api correctly and throw an error when site is not a content center', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts: any) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web?$select=WebTemplateConfiguration`) {
+        return {
+          WebTemplateConfiguration: 'SITEPAGEPUBLISHING#0'
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    try {
+      await spp.assertSiteIsContentCenter(siteUrl);
+      assert.fail('No error message thrown.');
+    }
+    catch (ex) {
+      assert.deepStrictEqual(ex, 'https://contoso.sharepoint.com is not a content site');
+    }
```

Comment:
Let's use `assert.rejects(...)`. It's also best to not throw plain text errors since this will increase the difficulty to write decent tests for it.
You can always take a look at other utils like `entraUser`, `entraGroup`, ...